nomad/client_csi_endpoint.go

```diff
@@ -354,7 +354,7 @@ func (a *ClientCSI) clientIDsForController(pluginID string) ([]string, error) {
 	// and have undocumented expectations of using k8s-specific sidecars to
 	// leader elect. Sort the client IDs so that we prefer sending requests to
 	// the same controller to hack around this.
-	clientIDs = sort.StringSlice(clientIDs)
+	slices.Sort(clientIDs)
```

Comment:
I was just caught out by this recently in a test I was writing. This is kind of a terrible API gotcha in the stdlib. Maybe we should add [SA4029](https://staticcheck.dev/docs/checks/#SA4029) to our staticcheck config?

Reply:
Good shout, I've added that here: https://github.com/hashicorp/nomad/pull/26640/commits/95713d945a9c8a1fd453587a914ce2b205da8b58

There are no other cases, so it felt OK sneaking it within this PR.